File: app/src/main/java/com/example/bomdiadeus/model/Verse.kt
package com.example.bomdiadeus.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.text.SimpleDateFormat
import java.util.*

/**
 * Classe modelo que representa um versículo bíblico.
 */
@Entity(tableName = "verses")
data class Verse(
    @PrimaryKey
    val dayOfYear: Int, // Dia do ano (1-365)
    val text: String,   // Texto do versículo
    val reference: String // Referência bíblica (ex: "João 3:16")
) {
    /**
     * Retorna a data formatada como "dd 'de' MMMM" com base no dia do ano.
     */
    fun getFormattedDate(): String {
        val calendar = Calendar.getInstance()
        val currentYear = calendar.get(Calendar.YEAR)
        calendar.set(currentYear, 0, 1)
        calendar.add(Calendar.DAY_OF_YEAR, dayOfYear - 1)
        
        val dateFormat = SimpleDateFormat("dd 'de' MMMM", Locale("pt", "BR"))
        return dateFormat.format(calendar.time)
    }
    
    /**
     * Retorna o mês (1-12) com base no dia do ano.
     */
    fun getMonth(): Int {
        val calendar = Calendar.getInstance()
        val currentYear = calendar.get(Calendar.YEAR)
        calendar.set(currentYear, 0, 1)
        calendar.add(Calendar.DAY_OF_YEAR, dayOfYear - 1)
        return calendar.get(Calendar.MONTH) + 1
    }
    
    /**
     * Retorna o dia do mês (1-31) com base no dia do ano.
     */
    fun getDayOfMonth(): Int {
        val calendar = Calendar.getInstance()
        val currentYear = calendar.get(Calendar.YEAR)
        calendar.set(currentYear, 0, 1)
        calendar.add(Calendar.DAY_OF_YEAR, dayOfYear - 1)
        return calendar.get(Calendar.DAY_OF_MONTH)
    }
    
    companion object {
        /**
         * Retorna o dia do ano (1-365) com base no mês e dia.
         */
        fun getDayOfYear(month: Int, day: Int): Int {
            val calendar = Calendar.getInstance()
            val currentYear = calendar.get(Calendar.YEAR)
            calendar.set(currentYear, month - 1, day)
            return calendar.get(Calendar.DAY_OF_YEAR)
        }
        
        /**
         * Retorna o dia do ano atual (1-365).
         */
        fun getCurrentDayOfYear(): Int {
            return Calendar.getInstance().get(Calendar.DAY_OF_YEAR)
        }
    }
} 

File: src/models/Verse.ts
/**
 * Modelo para um versículo bíblico
 */
export interface Verse {
  id: number;        // ID único (corresponde ao dia do ano, 1-365)
  text: string;      // Texto do versículo
  reference: string; // Referência bíblica (ex: "João 3:16")
}

/**
 * Funções utilitárias para manipulação de datas
 */
export const DateUtils = {
  /**
   * Obtém o dia do ano atual (1-365)
   */
  getCurrentDayOfYear: (): number => {
    const now = new Date();
    const start = new Date(now.getFullYear(), 0, 0);
    const diff = now.getTime() - start.getTime();
    const oneDay = 1000 * 60 * 60 * 24;
    const dayOfYear = Math.floor(diff / oneDay);
    return dayOfYear;
  },

  /**
   * Converte o dia do ano para uma data formatada
   */
  formatDayOfYear: (dayOfYear: number): string => {
    const year = new Date().getFullYear();
    const date = new Date(year, 0, dayOfYear);
    
    // Formatação em português (ex: "01 de janeiro")
    const day = date.getDate().toString().padStart(2, '0');
    const month = getMonthNameInPortuguese(date.getMonth());
    
    return `${day} de ${month}`;
  },

  /**
   * Obtém o dia do ano com base no mês e dia
   */
  getDayOfYear: (month: number, day: number): number => {
    const year = new Date().getFullYear();
    const date = new Date(year, month - 1, day);
    const start = new Date(year, 0, 0);
    const diff = date.getTime() - start.getTime();
    const oneDay = 1000 * 60 * 60 * 24;
    return Math.floor(diff / oneDay);
  }
};

/**
 * Retorna o nome do mês em português
 */
function getMonthNameInPortuguese(monthIndex: number): string {
  const months = [
    'janeiro',
    'fevereiro',
    'março',
    'abril',
    'maio',
    'junho',
    'julho',
    'agosto',
    'setembro',
    'outubro',
    'novembro',
    'dezembro'
  ];
  
  return months[monthIndex];
} 